components/server/src/analytics-controller.ts

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+import { inject, injectable } from "inversify";
+import express from "express";
+import {
+    IAnalyticsWriter,
+    IdentifyMessage,
+    RemoteIdentifyMessage,
+    RemotePageMessage,
+    RemoteTrackMessage,
+} from "@gitpod/gitpod-protocol/lib/analytics";
+import { createCookielessId, maskIp } from "./analytics";
+import { ClientHeaderFields, toClientHeaderFields } from "./express-util";
+import { Config } from "./config";
+import { RateLimited } from "./api/rate-limited";
+import { RateLimitter } from "./rate-limitter";
+import { RateLimiterRes } from "rate-limiter-flexible";
+
+@injectable()
+export class AnalyticsController {
```

Comment:
BrowserAnalyticsController - be specific that it is only for our browser clients